### PR TITLE
Add favorite/starred channels feature

### LINKED ISSUE #33: Implement favorite/starred/pinned channels
In the sidebar, provide a UI for setting channels as "favorites". This should keep the channel pinned to the top of the list of channels.

### PR DESCRIPTION
**Pull request by nikolaiwarner**
fixes #33

### CODE REVIEW

File: app/containers/mainPanel.js [hunk 1]
```diff
@@ -150,6 +158,19 @@ class MainPanel extends Component {
     })
   }
 
+  onToggleFavoriteChannel (channelName) {
+    let favorites = this.props.settings['favorite-channels'] || []
+    const index = favorites.indexOf(channelName)
+    if (index > -1) {
+      favorites.splice(index)
```
Comment (khubo) [inline app/containers/mainPanel.js:165]: maybe its better not to mutate props?

Comment (nikolaiwarner) [reply app/containers/mainPanel.js]: good call! thanks @khubo!

**Comment by cblgh** [comment]
😮

**Comment by cblgh** [comment]
i think collapsible channel listings are the unsung hero in this pr, soooooo nice!!

**Comment by nikolaiwarner** [comment]
thanks yall!

**Pull request merged**